### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by Emc2Ikeda**

### CODE REVIEW

File: src/main/java/StringManipulation.java [hunk 1]
```diff
@@ -1,32 +1,134 @@
...
+import java.util.Arrays;
+
 public class StringManipulation implements StringManipulationInterface {
+    String string = null;
+    // Default constructor
+    StringManipulation() {} 
+ 
+
+    // Parameterized constructor
+    StringManipulation(String string) {
```
Comment (haileymandal) [inline src/main/java/StringManipulation.java:11]: I see you have separate constructors, one default and other parameterized.  I was thinking you could merge and have one constructor.  What do you think?

File: src/main/java/StringManipulation.java [hunk 2]
```diff
@@ -1,32 +1,134 @@
...
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        // throw IllegalArgument Exception if n <= 0
+        if (n <= 0) {
+            throw new IllegalArgumentException("n has to be at least 1");
+        }
+
+        String string = this.getString();
```
Comment (haileymandal) [inline src/main/java/StringManipulation.java:47]: I see in the last method, you called getString();.  I think you could do the same here and do not need this.

File: src/main/java/StringManipulation.java [hunk 3]
```diff
@@ -1,32 +1,134 @@
...
-        return null;
+        return this.string;
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
+    // todo: raise nullPointer Exception if string is null
```
Comment (haileymandal) [inline src/main/java/StringManipulation.java:25]: I noticed this comment you had left for todo.  I thought I would share that this is a great idea for a test case.

**Review by haileymandal — APPROVED** [review]
Your test cases look great.  You did a few corner cases that I didn't think about.  One suggestion I have for them is changing the test case methods to show what you are doing vs comments above them.  

For example:  "public void removeNthCharacter_maintainSpacingFalse_returnsStringWithoutSpaces"

This is formatting and up to developers preference but I found it helpful when doing it myself.
